client/src/app/pages/migration-waves/migration-waves.tsx

```diff
@@ -160,10 +161,20 @@ export const MigrationWaves: React.FC = () => {
   );
 
   const removeApplication = (migrationWave: MigrationWave, id: number) => {
-    migrationWave.applications = migrationWave.applications.filter(
-      (application) => application.id !== id
+    const applicationRefs = toRefs(
+      migrationWave.applications.filter((application) => application.id !== id)
     );
-    updateMigrationWave(migrationWave);
+    const payload: MigrationWave = {
+      id: migrationWave.id,
+      name: migrationWave.name,
+      startDate: migrationWave.startDate,
+      endDate: migrationWave.endDate,
+      stakeholderGroups: migrationWave.stakeholderGroups,
+      stakeholders: migrationWave.stakeholders,
+      applications: applicationRefs || [],
+    };
```

Comment:
If most things are the same, then spreading the object and only specifying the keys to be overridden is a bit more compact:
```js
    const payload: MigrationWave = {
      ...migrationWave,
      applications: applicationRefs || [],
    };
```

or

```js
  updateMigrationWave({ ...payload, applications: applicationRefs || [] });
```

Reply:
I tried this, but there are some things included inthe migrationWave we have stored from the api which break the put request when submitting. So I explicitly defined what was passed here.

Reply:
So weird.  I wonder how many more json parse type errors we're going to hit this bug round.